doc/source/guidelines/version_support.rst

```diff
@@ -0,0 +1,38 @@
+Python Version Support
+======================
+
+When writing Python libraries, plan on supporting the oldest actively supported
+version of Python. For a quick reference, visit `Python EOL
+<https://endoflife.date/python>`_ summarized (as of 2022) here:
+
++---------+-------------+-----------------------+
+| Version | Released    | Security Support Ends |
++---------+-------------+-----------------------+
+| 3.10    | 04 Oct 2021 | 04 Oct 2026           |
++---------+-------------+-----------------------+
+| 3.9     | 05 Oct 2020 | 05 Oct 2025           |
++---------+-------------+-----------------------+
+| 3.8     | 14 Oct 2019 | 14 Oct 2024           |
++---------+-------------+-----------------------+
+| 3.7     | 27 Jun 2018 | 27 Jun 2023           |
++---------+-------------+-----------------------+
+
+Expect these to be the most commonly used Python versions. Note that recently dropped versions (Python 3.6) will no longer have wheels built for popular libraries like `numpy <https://numpy.org/>`_. Users can still install the older version from PyPI via ``pip``, and version support tends to gradually "drop off" rather than fall immediately when security support exits.
```

Comment:
```suggestion
Expect these to be the most commonly used Python versions. Note that recently dropped versions (Python 3.6) will no longer have wheels built for popular libraries like `numpy <https://numpy.org/>`_. You can still install the older version from PyPI via ``pip`` since version support tends to gradually "drop off" rather than cease entirely when security support exits.
```